### DISCUSSION TITLE
Cannot find symbol in Contract.LinkRefence web3j 4.8.9-android.

### DISCUSSION THREAD

**Discussion by johnspeny** [web3j/web3j#2091] [General]
Please update web3j android to tackle that issue especially when we add the generated contracts jnto our android projects. Unless there is a version of webcli that is specifically used for android version that am not aware of.

**Comment by rohandakua** [comment]
i am making the wrapper class from cli and the version there for web3j is 1.6.something . and in android it gemini says that it needs version o 4.~.~  . This contract.linkRefernce is the issue here . Are you facing the same issue . If not then please tell me how did you generate wrapper class for solidity file having version > then 4

**Reply by johnspeny** [reply]
Are you building for android? There is a specific version of cli that works well with this android version

**Reply by rohandakua** [reply]
yes i am building for android

**Reply by johnspeny** [reply]
Use webj-1.4.1
Its here [link](https://github.com/hyperledger/web3j-cli/releases)

Download and extract, i forgot which release specific one but the folder will be webj-1.4.1

**Reply by rohandakua** [reply]
sorry to say but there is no such version as 1.4.1

**Reply by johnspeny** [reply]
Not gona lie, try to download each of them and install since i forgot which 1 exactly.
When u type in the console.

web3j -v

It should show you version 1.4.1

**Reply by rohandakua** [reply]
is it possible to call methods of smart contract by java or kotlin in android project? if not then I would create a javascript server having apis that i will call from my android project. 
Have you done it before ?

**Reply by rohandakua** [reply]
but let me tell you , 1.4.1 version is present in web3j-cli  version but not in web3j 


web3j is having latest version of 4.something.something
web3j-cli has latest version of 1.6.something

**Reply by johnspeny** [reply]
Dont use the latest release of web3j-cli, look for older release that has the version i specified. I am sorry I forgot it. Start checking from v4.11.3 of the release going downwards. Then u need web3j 4.9.9-android added in ur gradle implementation.

```java 
implementation ('org.web3j:core:4.8.9-android') 
```

**Reply by johnspeny** [reply]
I tried the web3j way, but couldn't get past the method that checks eg helloWorld.isValid(). For checking whether a contract is wrong or not. I also found payable methods difficult to navigate. So I decided to go with your first approach and use metamask sdk on the client. It works smooth on my current project.

**Reply by johnspeny** [reply]
I have updated the link, this is what i meant.

Use webj-1.4.1 which gives u the cli
Its here [link](https://github.com/hyperledger/web3j-cli/releases)

Then in android use the implementation i pasted above

**Reply by rohandakua** [reply]
so i can do all this using metamask sdk also ? 

i will try in the evening and let you know about using cli 1.4.1

**Reply by johnspeny** [reply]
MetaMask sdk is just an option if u want your users to first connect to wallet then proceed using other smart contract methods u want them to access. Good luck

**Reply by rohandakua** [reply]
thanks for replying man , really i appreciate it 👍😁
